src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -119,7 +126,18 @@ public void dumpInventoryIntoWareHouse(@NotNull final InventoryCitizen inventory
             @Nullable final TileEntity chest = searchRightChestForStack(stack);
             if (chest == null)
             {
-                LanguageHandler.sendPlayersMessage(getColony().getMessagePlayerEntities(), COM_MINECOLONIES_COREMOD_WAREHOUSE_FULL);
+                if(world.getGameTime() - lastNotification > TICKS_FIVE_MIN)
+                {
+                    lastNotification = world.getGameTime();
+                    if(getBuilding().getBuildingLevel() == getBuilding().getMaxBuildingLevel())
+                    {
+                        LanguageHandler.sendPlayersMessage(getColony().getMessagePlayerEntities(), COM_MINECOLONIES_COREMOD_WAREHOUSE_FULL_MAX_UPGRADE);
```

Comment:
So, emerald block makes sense at level 5, but a warehouse full at level 4 this message is confusing.

Reply:
It should only give the "The Warehouse is full, please use an emerald block to make more space in the racks!" message at level five.  At level four, it will give "The Warehouse is full, please upgrade it!".  Did I goof that up?

I guess it might make sense to add a third message for when the warehouse is T5 and has all of its emerald upgrades,if that's the concern.